scrapy/utils/misc.py

```diff
@@ -226,7 +226,14 @@ def returns_none(return_node):
         return value is None or isinstance(value, ast.NameConstant) and value.value is None
 
     if inspect.isgeneratorfunction(callable):
-        code = re.sub(r"^[\t ]+", "", inspect.getsource(callable))
+        pattern = r"(^[\t ]+)"
+        src = inspect.getsource(callable)
+        match = re.match(pattern, src)  # Find indentation
+        code = re.sub(pattern, "", src)
+        if match:
+            # Remove indentation
+            code = re.sub(f"\n{match.group(0)}", "\n", code)
```

Comment:
Nitpick 2:
```suggestion
            code = re.sub(f"\n{match.group(0)}", "\n", code)  # remove indentation
```
More concise IMHO, and still respects the max line length